cassandra-diagnostics-reporter-prometheus/src/main/java/io/smartcat/cassandra/diagnostics/reporter/PrometheusReporter.java

```diff
@@ -0,0 +1,115 @@
+    private static final int DEFAULT_HTTP_SERVER_PORT = 9091;
+    private static final String ILLEGAL_CHARACTERS_REGEX = "-|\\.";
+    private static final String ALLOWED_DELIMITER = "_";
+
+    private static HTTPServer server;
+    final Map<String, Gauge> metricNameGuageMap = new HashMap<>();
+
+    /**
+     * Constructor.
+     *
+     * @param configuration Reporter configuration
+     * @param globalConfiguration Global configuration
+     * @throws IOException if HTTPServer cannot be initialized
+     */
+    public PrometheusReporter(ReporterConfiguration configuration, GlobalConfiguration globalConfiguration)
+            throws IOException {
+        super(configuration, globalConfiguration);
+        logger.debug("Initializing prometheus client with config: {}", configuration.toString());
+
+        final String httpServerHost = configuration.getOption(HTTP_SERVER_HOST_KEY);
+        final int httpServerPort = configuration.getDefaultOption(HTTP_SERVER_PORT_KEY, DEFAULT_HTTP_SERVER_PORT);
+
+        server = new HTTPServer(httpServerHost, httpServerPort);
+
+        logger.info("Initialized prometheus reporter on host and port: {}", httpServerHost + ":" + httpServerPort);
+    }
+
+    @Override
+    public void report(Measurement measurement) {
+        if (measurement.isSimple()) {
+            String name = measurement.name().replaceAll(ILLEGAL_CHARACTERS_REGEX, ALLOWED_DELIMITER);
+            Gauge gauge = getGaugeMeasurement(measurement, name);
+            String[] tagValues = convertTagValues(measurement.tags());
+            double value = measurement.getValue();
+            gauge.labels(tagValues).set(value);
+        } else {
+            String baseName = measurement.name().replaceAll(ILLEGAL_CHARACTERS_REGEX, ALLOWED_DELIMITER);
+            for (String key : measurement.fields().keySet()) {
+                if (!isNumeric(measurement.fields().get(key))) {
+                    continue;
+                }
+                String metricName = baseName + ":" + key.replaceAll(ILLEGAL_CHARACTERS_REGEX, ALLOWED_DELIMITER);
+                Gauge gauge = getGaugeMeasurement(measurement, metricName);
+                double value = Double.parseDouble(measurement.fields().get(key));
+                String[] tagValues = convertTagValues(measurement.tags());
+                gauge.labels(tagValues).set(value);
+            }
+        }
+    }
+
+    /**
+     * Get the Gauge object for the given measurementName from the metricNameGuageMap. If the object does not exist:
+     * create and register the metric and put it into the metricNameGuageMap.
+     *
+     * @param measurement
+     * @param measurementName parameter is added (instead of being fetched from Measurement object) because name of the
+     *            complex measurement is calculated for each field
+     * @return Gauge object for the given measurementName
+     */
+    private Gauge getGaugeMeasurement(final Measurement measurement, String measurementName) {
```

Comment:
Rename it to `getOrCreateGaugeMeasurement`. Took me a while to figure out where is the map handling.